presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -405,7 +406,22 @@ else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH) {
             ImmutableMap.Builder<TaskId, PagePartitionFunction> buffers = ImmutableMap.builder();
             for (int nodeIndex = 0; nodeIndex < parentTasks.size(); nodeIndex++) {
                 TaskId taskId = parentTasks.get(nodeIndex);
-                buffers.put(taskId, new HashPagePartitionFunction(nodeIndex, parentTasks.size(), getPartitioningChannels(fragment), getHashChannel(fragment), fragment.getTypes()));
+                buffers.put(taskId, new HashPagePartitionFunction(
```

Comment:
this change is no longer spurious after removing `HashNullEverywherePagePartitionFunction`